bin/build.js

```diff
@@ -196,12 +197,55 @@ export async function build() {
       );
     })
   );
+  await buildJsonFile();
   await freezeStaticHtml();
 }
 
 const copyStaticAssets = () =>
   step('📋 Copying static assets', () => fs.copy('static', 'dist/static'));
 
+const buildJsonFile = () => step('Building json file', () => buildJson());
+
+async function buildJson() {
+  let templateFolders = await fs.readdir('static/templates');
```

Comment:
1. The names are hardcoded below, which we don't want. You already have the names in this readdir, so let's just use it.

2. No need to do `new Array()`, just `[]`. Let's also create them inline.

3. Set a base constant for `static/templates`.

4. Nest loops for simplicity.

5. `endsWith('.html')` instead of `includes` for accuracy.

6. Name `files` instead for consistency.

7. [Avoid else, return early](https://blog.timoxley.com/post/47041269194/avoid-else-return-early)

8. Use [`Array.filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) for simplicity.

9. Ignore system files.

10. Ignore files on templates root that aren't directories.

11. `fs.outputJson` for simplicity.

12. Just pass `fatal` to `catch`.

```js
const buildTemplatesJson = step('Generating templates json', async () => {
  const root = 'static/templates';
  const result = {};
  for (const name of await fs.readdir(root)) {
    const fileUrl = file => `${root}/${name}/${file}`;
    const preview = {type: 'image'};
    let contentUrl;
    let files;
    try {
      files = await fs.readdir(`${root}/${name}`);
    } catch(_) {
      // not a directory, not a template.
      continue;
    }
    result[name] = {
      preview,
      files: files.filter(file => {
        if (file.endsWith('.html')) {
          contentUrl = fileUrl(file);
          return false;
        }
        if (file.startsWith('_preview.') {
          preview.url = fileUrl(file);
          if (file.endsWith('.mp4')) {
            preview.type = 'video';
          }
          return false;
        }
        return !file.startsWith('.');
      },
      // (this needs to be at the end since we're setting it above)
      contentUrl,
    });
  }
  return fs.outputJson('dist/templates.json', result).catch(fatal);
});
```